### PR TITLE
[Merged by Bors] - feat: Remove `Extensive` in favour of `FinitaryExtensive`.

### PR DESCRIPTION
**Pull request by erdOne**
---

Not 100% sure this change is desired. `FinitaryExtensive` is stronger than `Extensive` as the former requires furthermore that coproducts satisfy some form of descent.
However the former is what the literature uses and has some better properties (e.g. coproducts are disjoint).
Nonetheless the results in `Mathlib/CategoryTheory/Extensive.lean` should be valuable regardless.

Also, the aforementioned file is a bit too long and will be split in a future PR.

- [x] depends on: #7745

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Sites/RegularExtensive.lean [hunk 1]
```diff
@@ -67,39 +68,6 @@ class Preregular : Prop where
...
-  HasPullbacksOfInclusions C := ⟨fun _ _ _ => inferInstance⟩
-
-/--
-A category is *extensive* if it has all finite coproducts and those coproducts are preserved
-by pullbacks (we only require the relevant pullbacks to exist, via `HasPullbacksOfInclusions`).
-
-TODO: relate this to the class `FinitaryExtensive`
--/
-class Extensive extends HasFiniteCoproducts C, HasPullbacksOfInclusions C : Prop where
-  /-- Pulling back an isomorphism from a coproduct yields an isomorphism. -/
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Sites/RegularExtensive.lean:98]: I would like this class to remain, under the name `Preextensive`/`PreFinitaryExtensive`. Then we can leave this file as it is and just add an `instance [FinitaryExtensive C] : Preextensive C`

**Comment by dagurtomas** [comment]
Aside from my other comment, I'm very happy with the additions of this PR. Thanks!

**Comment by erdOne** [comment]
@dagurtomas Does this implementation (following [this comment of mine](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Extensive.20Coverage/near/397266476)) look good to you?

**Comment by dagurtomas** [comment]
Yes, I'm happy with this. Thank you!

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

**Comment by erdOne** [comment]
bors merge

**Comment by erdOne** [comment]
bors merge

**Pull request closed without merging**